jasmine/src/roll.js

```diff
@@ -0,0 +1,19 @@
+function Roll(game, frame){
+  this.game = game;
+  this.frameScore = null;
+  this.isStrike = false;
+  this.isSpare = false;
+  this.scoreSoFar = 0;
+};
+
+Roll.prototype.roll = function(score1, score2) {
+  if (score1 == 10) { this.isStrike = true; } //strike
+  else if (score2 + score1 == 10){ this.isSpare = true; } // spare
+  this.addScore(score1, score2, this.isStrike, this.isSpare);
+};
+
+Roll.prototype.addScore = function (score1, score2, strikeStatus, spareStatus) {
+  this.scoreSoFar += (score1 + score2); // total for frame (before any bonuses added)
+  this.frameScore = {'Ball 1: ' : score1, 'Ball 2:' : score2, 'Strike: ' : strikeStatus, 'Spare: ' : spareStatus }; // push values to Frame Score array
```

Comment:
Whenever you find yourself using a hash in a case where you are likely to have more than one instance and those instances will have a consistent data structure, think about using a class instead. 

Having a class usually (a) makes your code easier to follow, but also (b) gives you a mechanism for encapsulation. For example, you may decide that it is the responsible of a frame class (rather than a roll) to know whether it is a strike or a spare, so you could have a method on the frame:

```
Frame.prototype.isStrike = function() {
   return this.score1 === 10
}
Frame.prototype.isSpare = function() {
   return !this.isStrike() && (this.score1 + this.score2 === 10)
}
```

This refactor isn't necessary, this is just for illustration - injecting the strike and spare values from a roll class is also a perfectly reasonable approach. However, if you were to start integrating this with a UI, I suspect that you would start seeing the value of being able to attach logic to an instance of the frame class rather than just having an array of hashes.